### DISCUSSION TITLE
RecipeVariantProps gives undefined in unions due to which storybook doesn't render controls

### DISCUSSION THREAD

**Discussion by abhayzmp** [chakra-ui/panda#2771] [General]
### Description

When I use below to export types
```tsx
type ButtonVariantProps = NonNullable<RecipeVariantProps<typeof buttonRecipe>>;
```

Output is 
```tsx
type ButtonVariantProps = {
    variant?: "outline" | "solid" | undefined;
    size?: "medium" | "large" | undefined;
    shape?: "rounded" | "pill" | "square" | undefined;
}
```
and storybook doesn't render controls
![image](https://github.com/user-attachments/assets/a3fb8ccd-9b8d-4b22-a2c3-e2f5b24dd2da)

But when I pass the types like below
```tsx
export type ButtonProps = ComponentProps<"button"> & {
  variant?: "outline" | "solid";
  size?: "medium" | "large";
  shape?: "rounded" | "pill" | "square";
};
```

Storybook renders the controls correctly
![image](https://github.com/user-attachments/assets/b62f8c31-1dae-4840-86bc-b5b8e359b674)

I was just following this tutorial
https://www.youtube.com/watch?v=KgcSZednCmk

### Link to Reproduction

https://www.youtube.com/watch?v=KgcSZednCmk

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### JS Framework

React (TS)

### Panda CSS Version

0.44.0

### Browser

Version 127.0.6533.89 (Official Build) (arm64)

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by abhayzmp** [comment]
Tried this, didn't work
```tsx

type WithoutUndefined<T> = {
    [P in keyof T]?: T[P];
};

type ButtonVariantProps = WithoutUndefined<{
    variant: "outline" | "solid";
    size: "medium" | "large";
    shape: "rounded" | "pill" | "square";
}>;
```

**Reply by cgatian** [reply]
`NonNullable<ButtonVariantProps>` might be what youre looking for
